WordPress/src/main/java/org/wordpress/android/ui/prefs/DotComSiteSettings.java

```diff
@@ -268,6 +276,19 @@ private void deserializeDotComRestResponse(SiteModel site, JSONObject response)
             mRemoteSettings.sortCommentsBy = DESCENDING_SORT;
         }
 
+        JSONObject jetpackProtectWhitelist = settingsObject.optJSONObject(JP_PROTECT_WHITELIST_KEY);
+        if (jetpackProtectWhitelist != null) {
+            JSONArray whitelistItems = jetpackProtectWhitelist.optJSONArray("local");
+            if (whitelistItems != null) {
+                for (int i = 0; i < whitelistItems.length(); ++i) {
+                    String item = whitelistItems.optString(i, "");
+                    if (!item.isEmpty() && !mRemoteJpSettings.jetpackProtectWhitelist.contains(item)) {
+                        mRemoteJpSettings.jetpackProtectWhitelist.add(i, item);
```

Comment:
I am just curious, do we really need to add the item to a specific position? If we keep using List, I think this has the potential to break. Consider the case where we get 5 items and the 3rd item fails the `isEmpty` check (assuming that's possible since we have the check for it), then if we add items to specific positions, we'll have a `null` in the 3rd position right? Not 100% sure, but I think that'll be the case.